### PR TITLE
[8.x] Specify more concrete data types inside of arrays in PHPDoc

### PR DESCRIPTION
**Pull request by antonkomarev**
This PR specifies type of data inside arrays. It affects only the most obvious arrays to make review as simple as possible.

### CODE REVIEW

File: tests/Integration/Database/DatabaseEloquentModelCustomCastingTest.php [hunk 1]
```diff
@@ -239,14 +239,14 @@ class TestEloquentModelWithCustomCast extends Model
...
-     * @var array
+     * @var string[]
      */
     protected $guarded = [];
 
     /**
      * The attributes that should be cast to native types.
      *
-     * @var array
+     * @var string[]
```
Comment (GrahamCampbell) [inline tests/Integration/Database/DatabaseEloquentModelCustomCastingTest.php:249]: wrong

File: tests/Integration/Auth/Fixtures/Models/AuthenticationTestUser.php [hunk 2]
```diff
@@ -12,14 +12,14 @@ class AuthenticationTestUser extends Authenticatable
...
-     * @var array
+     * @var string[]
      */
     protected $guarded = [];
 
     /**
      * The attributes that should be hidden for arrays.
      *
-     * @var array
+     * @var string[]
```
Comment (GrahamCampbell) [inline tests/Integration/Auth/Fixtures/Models/AuthenticationTestUser.php:22]: wrong

File: tests/Integration/Auth/Fixtures/AuthenticationTestUser.php [hunk 3]
```diff
@@ -12,14 +12,14 @@ class AuthenticationTestUser extends Authenticatable
...
-     * @var array
+     * @var string[]
      */
     protected $guarded = [];
 
     /**
      * The attributes that should be hidden for arrays.
      *
-     * @var array
+     * @var string[]
```
Comment (GrahamCampbell) [inline tests/Integration/Auth/Fixtures/AuthenticationTestUser.php:22]: wrong. `string[]` is an alias for `array<int,string>` but this type is `array<string,string>`.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: tests/Integration/Auth/Fixtures/AuthenticationTestUser.php (hunk 3 above)
Comment (antonkomarev) [reply tests/Integration/Auth/Fixtures/AuthenticationTestUser.php:22]: I agree that some of my fixes are wrong, but this one doesn't look like wrong.

```php
protected $hidden = [
        'password', 'remember_token',
];
```
Is equivalent for:
```php
protected $hidden = [
        0 => 'password', 1 => 'remember_token',
];
```

File: tests/Integration/Database/DatabaseEloquentModelCustomCastingTest.php (hunk 1 above)
Comment (antonkomarev) [reply tests/Integration/Database/DatabaseEloquentModelCustomCastingTest.php]: Fixed

File: tests/Integration/Auth/Fixtures/Models/AuthenticationTestUser.php (hunk 2 above)
Comment (antonkomarev) [reply tests/Integration/Auth/Fixtures/Models/AuthenticationTestUser.php:22]: not wrong change

**Comment by antonkomarev** [comment]
Thanks @GrahamCampbell I've fixed all issues in this PR. Now only `array<int, string>` arrays has `string[]` type

**Pull request merged**